spark/src/test/java/org/apache/iceberg/spark/data/TestHelpers.java

```diff
@@ -267,22 +267,24 @@ private static void assertEqualsUnsafe(Type type, Object expected, Object actual
     }
 
     switch (type.typeId()) {
-      case INTEGER:
-        Assert.assertEquals("Values didn't match", ((Number) expected).intValue(),
-                ((Number) actual).intValue());
-        break;
       case LONG:
-        Assert.assertEquals("Values didn't match", ((Number) expected).longValue(),
-                ((Number) actual).longValue());
-        break;
-      case FLOAT:
-        Assert.assertEquals("Values didn't match", Float.floatToIntBits(((Number) expected).floatValue()),
-                Float.floatToIntBits(((Number) actual).floatValue()));
+        if (expected instanceof Integer) {
+          Assert.assertEquals("Values didn't match", ((Number) expected).longValue(),
+              ((Number) actual).longValue());
```

Comment:
This should not alter the type of the actual value. If actual is an `int`, that's still an error.